### DISCUSSION TITLE
Replicator Configuration in Dev 3.0.0 Version

### DISCUSSION THREAD

**Discussion by Konst80** [cofu-app/cbl-dart#534] [Q&A] [answered]
Hello everyone,

I have a question concerning the new (dev) 3.0.0 version in the context of the replicator.

Utilizing the code below leads to the following message:
```
'database' is deprecated and shouldn't be used. Add a CollectionConfiguration for the default collection instead.
Try replacing the use of the deprecated member with the replacement.
```
Code:
```
 final replicatorConfiguration = ReplicatorConfiguration(
        database: database,
        enableAutoPurge: true,
        continuous: true,
        replicatorType: ReplicatorType.pushAndPull,
        authenticator: SessionAuthenticator(sessionId: sessionID),
        target: UrlEndpoint(Uri.parse(CouchbaseConfig.syncUrl)));
```

My question is: how should the new function be used in this context? Could someone provide an example, please?

`final collectionConfiguration = CollectionConfiguration(channels: channels)
`

**Comment by blaugold** [comment, marked as answer]
The way to configure a collection for replication is through [`addCollection`](https://pub.dev/documentation/cbl/3.0.0-dev.1/cbl/ReplicatorConfiguration/addCollection.html) or [`addCollections`](https://pub.dev/documentation/cbl/3.0.0-dev.1/cbl/ReplicatorConfiguration/addCollections.html).

```dart
 final replicatorConfiguration = ReplicatorConfiguration(
        enableAutoPurge: true,
        continuous: true,
        replicatorType: ReplicatorType.pushAndPull,
        authenticator: SessionAuthenticator(sessionId: sessionID),
        target: UrlEndpoint(Uri.parse(CouchbaseConfig.syncUrl))
);
replicatorConfiguration.addCollection(database.defaultCollection, CollectionConfiguration(channels: channels));
```
